### PR TITLE
Added support for capacity and profile

### LINKED ISSUE #74: Support for images that require larger than 100G boot volume
**Describe the bug**
Capability to provide custom boot volume size

### PR DESCRIPTION
**Pull request by BShrivastav**
**Supported Feature:**

-  Boot Volume Capacity
-  Boot Volume Profile
Closes #74

### CODE REVIEW

File: builder/ibmcloud/vpc/config.go [hunk 1]
```diff
@@ -84,6 +86,14 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, error) {
 		errs = packer.MultiErrorAppend(errs, errors.New("a subnet_id must be specified"))
 	}
 
+	if c.VSIBootCapacity != 0 && (c.VSIBootCapacity < 100 || c.VSIBootCapacity > 250) {
+		errs = packer.MultiErrorAppend(errs, errors.New("boot capacity out of bound: provide a valid capacity between 100 to 250"))
+	}
+	if c.VSIVolProfile != "" && (c.VSIVolProfile != "5iops-tier" || c.VSIVolProfile != "10iops-tier" || c.VSIVolProfile != "custom" || c.VSIVolProfile != "general-purpose") {
+		// errs = packer.MultiErrorAppend(errs, errors.New("Volume profile can be one of: general-purpose, 5iops-tier, 10iops-tier or custom"))
```
Comment (ujjwal-ibm) [inline builder/ibmcloud/vpc/config.go:93]: remove commented code

Comment (BShrivastav) [reply builder/ibmcloud/vpc/config.go]: done.

**Comment by astha-jain** [comment]
Please add developer examples and examples

File: developer/examples/build.vpc.catalogimage-offering.pkr.hcl [hunk 2]
```diff
@@ -55,6 +55,9 @@ source "ibmcloud-vpc" "catalog" {
   vsi_interface      = "public"
   vsi_user_data_file = ""
 
+  vsi_boot_vol_capacity = "200"
```
Comment (astha-jain) [inline developer/examples/build.vpc.catalogimage-offering.pkr.hcl:58]: Just one example file is enough. You don't need to add this change to every example.

File: README.md [hunk 3]
```diff
@@ -202,6 +202,8 @@ vsi_interface | string | Set it as "public" to create a Floating IP to connect t
 vsi_user_data_file | string | User data to be made available when setting up the virtual server instance. Optional.
 | OR |
 vsi_user_data | string | User data to be made available when setting up the virtual server instance. Optional. This is the string input variable.
+vsi_boot_vol_capacity | string | The capacity to use for the volume (in gigabytes). Must be at least the image's minimum_provisioned_size. The maximum value may increase in the future. Optional.
+vsi_boot_vol_profile | string | User can provide the available profile for volume attachments. Refer https://cloud.ibm.com/docs/vpc?topic=vpc-block-storage-profiles&interface=ui for profile info. Optional.
```
Comment (astha-jain) [inline README.md:206]: list of allowed values needs to be mentioned here. Document does not list the allowed values

File: developer/examples/build.vpc.windows.pkr.hcl [hunk 4]
```diff
@@ -50,6 +50,9 @@ source "ibmcloud-vpc" "windows" {
   vsi_interface       = "public"
   vsi_user_data_file  = "scripts/winrm_setup.ps1"
 
+  vsi_boot_vol_capacity = "200"
+  vsi_boot_vol_profile  = "5iops-tier"
+
```
Comment (astha-jain) [inline developer/examples/build.vpc.windows.pkr.hcl:55]: You can remove the two from develop examples and only keep examples.

**Comment by BShrivastav** [comment]
Use this PR https://github.com/IBM/packer-plugin-ibmcloud/pull/83

**Pull request closed without merging**